### PR TITLE
Add better error message for trace infinite loop

### PR DESCRIPTION
**Pull request by leonardt**

### CODE REVIEW

File: tests/test_wire/test_wireable.py [hunk 1]
```diff
@@ -20,3 +21,22 @@ class Main2(m.Circuit):
 Cannot wire Main2.a (Out(SInt[16])) to Main2.b (In(UInt[16]))\
 """
     assert caplog.messages[1][-len(expected):] == expected
+
+
+def test_trace_infinite_loop():
+    class Foo(m.Circuit):
```
Comment (rsetaluri) [inline tests/test_wire/test_wireable.py:27]: we can do this without a circuit right? if we do need the circuit wrapper for some reason let's name it `_`

File: magma/wire_container.py [hunk 2]
```diff
@@ -201,7 +210,10 @@ def wired(self):
 
     # return the input or output Bit connected to this Bit
     def trace(self, skip_self=True):
-        return self._wire.trace(skip_self)
+        try:
```
Comment (rsetaluri) [inline magma/wire_container.py:213]: I would say this is an ok way to do it, but perhaps needs better logic to actually detect a combo loop. not sure the best way to that...

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -21,6 +21,10 @@ jobs:
       run: |
         sudo apt install -y verilator libgmp-dev libmpfr-dev libmpc-dev
         python -m pip install --upgrade pip
+        # BEGIN: Temp fix, see
```
Comment (rsetaluri) [inline .github/workflows/deploy.yml:24]: needs a rebase?

**Review by rsetaluri — APPROVED** [review]
looks good, please address comments before merging

File: magma/wire_container.py (hunk 2 above)
Comment (leonardt) [reply magma/wire_container.py:213]: Actually a more precise way might be to keep a list of the current trace and error if we encounter a seen value, will try that.

Comment (rsetaluri) [reply magma/wire_container.py:213]: To be fair, I don't think this is a bad way to do this. Letting the system take care of tracking those values for us...

Comment (leonardt) [reply magma/wire_container.py:213]: It technically requires more stack frames than necessary (e.g. we could catch the loop in three recursive calls versus waiting for the RecursionError), but now that I think about it, given this is the very uncommon case, it doens't seem worth it to add extra cost here just for that error.

File: magma/wire_container.py [hunk 4]
```diff
@@ -12,6 +12,15 @@
 )
 
 
+class TraceRecursionError(RecursionError):
+    def __init__(self, value):
+        msg = (
+            f"RecursionError when calling trace on {value}, do you have a"
+            " combinational loop?"
```
Comment (rsetaluri) [inline magma/wire_container.py:19]: nit```suggestion
            f"RecursionError when calling trace on {value}, do you have a "
            "combinational loop?"
```

**Review by rsetaluri — APPROVED** [review]
last nit

**Pull request merged**